### PR TITLE
doc: update upgrade guide

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

Ref:

* #824 

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: docs/en/latest/upgrade.md [hunk 1]
```diff
@@ -21,6 +21,77 @@ title: Upgrade Guide
...
+  --set ingress-controller.enabled=true \
+  --namespace ingress-apisix \
+  --set ingress-controller.config.apisix.serviceNamespace=ingress-apisix
+```
+
+### Precautions
+
+It is recommended not to upgrade across major versions.
+
+#### 1.4 -> 1.5 -> 1.6
```
Comment (lingsamuel) [inline docs/en/latest/upgrade.md:71]: What about `1.4 to 1.6`? We just need to know the minimum and maximum versions.

File: docs/en/latest/upgrade.md [hunk 2]
```diff
@@ -21,6 +21,77 @@ title: Upgrade Guide
...
+
+Compatible with upgrades without changing any resources.
+
+#### 1.3 -> 1.4
```
Comment (lingsamuel) [inline docs/en/latest/upgrade.md:75]: On my side, I would like the incompatible version to have a separate section, and the compatible version to have another one. Don't mix them together.

Comment (AlinsRan) [reply docs/en/latest/upgrade.md:75]: You're right.

File: docs/en/latest/upgrade.md (hunk 1 above)
Comment (tao12345666333) [reply docs/en/latest/upgrade.md:71]: no, I don't think we can upgrade to v1.6 from v1.4 directly.

File: docs/en/latest/upgrade.md (hunk 2 above)
Comment (tao12345666333) [reply docs/en/latest/upgrade.md:75]: +1

File: docs/en/latest/upgrade.md [hunk 3]
```diff
@@ -21,6 +21,77 @@ title: Upgrade Guide
...
+## Version change
+
+### 1.6.0
+
+- No breaking changes in this release.
+
+### 1.5.0
+
+- CRD has been upgraded to the V2 version, and V2beta3 has been marked as deprecated.
+- app_namespace e is deprecated, you can use namespace_selector instead.
```
Comment (tao12345666333) [inline docs/en/latest/upgrade.md:89]:
```suggestion
- app_namespace is deprecated, you can use namespace_selector instead.
```

**Review by lingsamuel — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**